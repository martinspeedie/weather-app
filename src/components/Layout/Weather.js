import React from 'react';
import tw from 'twin.macro';
import { connect } from 'react-redux';

// Components
import WeatherCard from '../Elements/WeatherCard';

// Styles
const Wrapper = tw.div`text-center`;

const Title = tw.p`inline-block text-3xl mt-12 lg:(mt-0)`;

const Location = tw.span`text-yellow-500`;

const WeatherWrapper = tw.div`grid mt-6 gap-y-3 grid-cols-1 lg:(grid-cols-6 gap-x-3)`;

export const Weather = ({ searchData }) => {
    const weather = searchData.weather;
    if (!weather) return null;
    const forecast = weather.consolidated_weather?.slice(0, 5);

    return (
        <Wrapper>
            <Title>
                5 day forecast for <Location>{weather.title}</Location>
            </Title>
            <WeatherWrapper>
                {forecast.map((weather) => (
                    <WeatherCard key={weather.id} data={weather} />
                ))}
            </WeatherWrapper>
        </Wrapper>
    );
};

const mapStateToProps = ({ searchData }) => ({ searchData });

export default connect(mapStateToProps)(Weather);
