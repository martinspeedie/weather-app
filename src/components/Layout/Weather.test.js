import 'jsdom-global/register';
import toJson from 'enzyme-to-json';

// Local imports
import '../../setupTests';

// Components
import { Weather } from './Weather';

describe('The Weather Component', () => {
    const mockData = {
        weather: {
            title: 'Melbourne',
            consolidated_weather: [
                {
                    id: 5482111432654848,
                    weather_state_name: 'Showers',
                    weather_state_abbr: 's',
                    wind_direction_compass: 'ENE',
                    created: '2021-09-28T22:24:55.860839Z',
                    applicable_date: '2021-09-29',
                    min_temp: 17.05,
                    max_temp: 21.595,
                    the_temp: 25.205,
                    wind_speed: 4.660419734658926,
                    wind_direction: 75.91824591867677,
                    air_pressure: 1017,
                    humidity: 62,
                    visibility: 14.885568281237571,
                    predictability: 73,
                },
                {
                    id: 6457784498388992,
                    weather_state_name: 'Showers',
                    weather_state_abbr: 's',
                    wind_direction_compass: 'E',
                    created: '2021-09-28T22:24:58.747855Z',
                    applicable_date: '2021-09-30',
                    min_temp: 18.15,
                    max_temp: 23.02,
                    the_temp: 23.265,
                    wind_speed: 5.346548062570588,
                    wind_direction: 91.29229524054001,
                    air_pressure: 1016,
                    humidity: 77,
                    visibility: 8.362724190726158,
                    predictability: 73,
                },
                {
                    id: 4946643204964352,
                    weather_state_name: 'Heavy Rain',
                    weather_state_abbr: 'hr',
                    wind_direction_compass: 'SE',
                    created: '2021-09-28T22:25:02.006956Z',
                    applicable_date: '2021-10-01',
                    min_temp: 18.085,
                    max_temp: 23.18,
                    the_temp: 24.015,
                    wind_speed: 4.676773429083486,
                    wind_direction: 130.95471384523373,
                    air_pressure: 1013,
                    humidity: 76,
                    visibility: 10.702497415095841,
                    predictability: 77,
                },
                {
                    id: 5160516495343616,
                    weather_state_name: 'Showers',
                    weather_state_abbr: 's',
                    wind_direction_compass: 'SW',
                    created: '2021-09-28T22:25:04.159475Z',
                    applicable_date: '2021-10-02',
                    min_temp: 18.02,
                    max_temp: 22.71,
                    the_temp: 23.884999999999998,
                    wind_speed: 4.862840007025637,
                    wind_direction: 226.28328323481168,
                    air_pressure: 1013.5,
                    humidity: 73,
                    visibility: 11.262974230493915,
                    predictability: 73,
                },
                {
                    id: 6503845204066304,
                    weather_state_name: 'Light Rain',
                    weather_state_abbr: 'lr',
                    wind_direction_compass: 'ESE',
                    created: '2021-09-28T22:25:07.358976Z',
                    applicable_date: '2021-10-03',
                    min_temp: 17.455,
                    max_temp: 23.365000000000002,
                    the_temp: 23.685,
                    wind_speed: 5.161674952681673,
                    wind_direction: 101.95027547375837,
                    air_pressure: 1012.5,
                    humidity: 74,
                    visibility: 9.271168873777142,
                    predictability: 75,
                },
                {
                    id: 5114351502491648,
                    weather_state_name: 'Showers',
                    weather_state_abbr: 's',
                    wind_direction_compass: 'E',
                    created: '2021-09-28T22:25:10.175389Z',
                    applicable_date: '2021-10-04',
                    min_temp: 17.175,
                    max_temp: 22.665,
                    the_temp: 27.34,
                    wind_speed: 5.981726318301122,
                    wind_direction: 100.50000000000001,
                    air_pressure: 1016,
                    humidity: 66,
                    visibility: 9.944424560566294,
                    predictability: 73,
                },
            ],
        },
    };
    it('renders correctly', () => {
        const tree = shallow(<Weather searchData={mockData} />);
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('displays the city name', () => {
        const wrapper = mount(<Weather searchData={mockData} />);
        const value = wrapper.find('span').first().text();
        expect(value).toEqual(mockData.weather.title);
    });
});
