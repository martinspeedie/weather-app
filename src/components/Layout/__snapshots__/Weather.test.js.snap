// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Weather Component renders correctly 1`] = `
<Weather__Wrapper>
  <Weather__Title>
    5 day forecast for 
    <Weather__Location>
      Melbourne
    </Weather__Location>
  </Weather__Title>
  <Weather__WeatherWrapper>
    <WeatherCard
      data={
        Object {
          "air_pressure": 1017,
          "applicable_date": "2021-09-29",
          "created": "2021-09-28T22:24:55.860839Z",
          "humidity": 62,
          "id": 5482111432654848,
          "max_temp": 21.595,
          "min_temp": 17.05,
          "predictability": 73,
          "the_temp": 25.205,
          "visibility": 14.885568281237571,
          "weather_state_abbr": "s",
          "weather_state_name": "Showers",
          "wind_direction": 75.91824591867677,
          "wind_direction_compass": "ENE",
          "wind_speed": 4.660419734658926,
        }
      }
      key="5482111432654848"
    />
    <WeatherCard
      data={
        Object {
          "air_pressure": 1016,
          "applicable_date": "2021-09-30",
          "created": "2021-09-28T22:24:58.747855Z",
          "humidity": 77,
          "id": 6457784498388992,
          "max_temp": 23.02,
          "min_temp": 18.15,
          "predictability": 73,
          "the_temp": 23.265,
          "visibility": 8.362724190726158,
          "weather_state_abbr": "s",
          "weather_state_name": "Showers",
          "wind_direction": 91.29229524054001,
          "wind_direction_compass": "E",
          "wind_speed": 5.346548062570588,
        }
      }
      key="6457784498388992"
    />
    <WeatherCard
      data={
        Object {
          "air_pressure": 1013,
          "applicable_date": "2021-10-01",
          "created": "2021-09-28T22:25:02.006956Z",
          "humidity": 76,
          "id": 4946643204964352,
          "max_temp": 23.18,
          "min_temp": 18.085,
          "predictability": 77,
          "the_temp": 24.015,
          "visibility": 10.702497415095841,
          "weather_state_abbr": "hr",
          "weather_state_name": "Heavy Rain",
          "wind_direction": 130.95471384523373,
          "wind_direction_compass": "SE",
          "wind_speed": 4.676773429083486,
        }
      }
      key="4946643204964352"
    />
    <WeatherCard
      data={
        Object {
          "air_pressure": 1013.5,
          "applicable_date": "2021-10-02",
          "created": "2021-09-28T22:25:04.159475Z",
          "humidity": 73,
          "id": 5160516495343616,
          "max_temp": 22.71,
          "min_temp": 18.02,
          "predictability": 73,
          "the_temp": 23.884999999999998,
          "visibility": 11.262974230493915,
          "weather_state_abbr": "s",
          "weather_state_name": "Showers",
          "wind_direction": 226.28328323481168,
          "wind_direction_compass": "SW",
          "wind_speed": 4.862840007025637,
        }
      }
      key="5160516495343616"
    />
    <WeatherCard
      data={
        Object {
          "air_pressure": 1012.5,
          "applicable_date": "2021-10-03",
          "created": "2021-09-28T22:25:07.358976Z",
          "humidity": 74,
          "id": 6503845204066304,
          "max_temp": 23.365000000000002,
          "min_temp": 17.455,
          "predictability": 75,
          "the_temp": 23.685,
          "visibility": 9.271168873777142,
          "weather_state_abbr": "lr",
          "weather_state_name": "Light Rain",
          "wind_direction": 101.95027547375837,
          "wind_direction_compass": "ESE",
          "wind_speed": 5.161674952681673,
        }
      }
      key="6503845204066304"
    />
  </Weather__WeatherWrapper>
</Weather__Wrapper>
`;
